observability/src/test/java/io/grpc/observability/logging/GcpLogSinkTest.java

```diff
@@ -84,6 +87,41 @@ public void verifyWrite() throws Exception {
     verifyNoMoreInteractions(mockLogging);
   }
 
+  @Test
+  @SuppressWarnings("unchecked")
+  public void verifyWriteWithTags() {
+    GcpLogSink mockSink = new GcpLogSink(mockLogging);
+    Map<String, String> locationTags = ImmutableMap.of("project_id", "PROJECT",
+        "location", "us-central1-c",
+        "cluster_name", "grpc-observability-cluster",
+        "namespace_name", "default" ,
+        "pod_name", "app1-6c7c58f897-n92c5");
+    Map<String, String> customTags = ImmutableMap.of("KEY1", "Value1",
+        "KEY2", "VALUE2");
+    GrpcLogRecord logProto = GrpcLogRecord.newBuilder()
+        .setRpcId("1234")
+        .build();
+
+    MonitoredResource expectedMonitoredResource = mockSink.getResource(locationTags);
+    Struct expectedStructLogProto = Struct.newBuilder().putFields(
+        "rpc_id", Value.newBuilder().setStringValue("1234").build()
+    ).build();
+
+    mockSink.write(logProto, locationTags, customTags);
```

Comment:
If we set the locationTags, customTags in GcpLogSink just once then the class and the tests will change

Reply:
Updated tests as well.